src/main/kotlin/com/sourcegraph/cody/agent/protocol/ContextItem.kt

```diff
@@ -1,25 +1,51 @@
 package com.sourcegraph.cody.agent.protocol
 
-import com.google.gson.*
+import com.google.gson.JsonDeserializationContext
+import com.google.gson.JsonDeserializer
+import com.google.gson.JsonElement
+import com.google.gson.JsonNull
+import com.google.gson.JsonObject
+import com.google.gson.JsonSerializer
 import java.lang.reflect.Type
 import java.net.URI
 import java.nio.file.Path
 import java.nio.file.Paths
 
-sealed class ContextFile {
-  abstract val type: String
+typealias ContextFileSource =
+    String // One of: embeddings, user, keyword, editor, filename, search, unified, selection,
+// terminal
+
+typealias SymbolKind = String // One of: class, function, method
+
+sealed class ContextItem {
+  abstract val type: String // Oneof: file, symbol
   abstract val uri: URI
```

Comment:
The generated protocol is not perfect. In particular `ContextItem` contains no fields in it while these two can and should be included here. 

I think that in general the binding script and the generated binding are helpful but we must carefully review them.